### ISSUE TITLE
Functional version of torch.nn.LSTM -> torch.nn.functional.LSTM

### ISSUE THREAD

**Issue by mariaalfaroc** [pytorch/pytorch#80454]
### 🚀 Requesting torch.nn.Functional.LSTM

I am diving into meta-learning and need to change the weights of module during the training process, so I can’t use off-the-shelf `torch.nn.Conv2d` or `torch.nn.LSTM` module as I can’t pass weights into the module. 

For convolution layers or batch normalization layers, PyTorch provides `torch.nn.Functional.conv2d` and `torch.nn.Functional.batchnorm` interface, and can be called easily. Things are a little different for LSTM module, there is no interface like `torch.nn.Functional.LSTM`.

It would be great to implement one!

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by albanD** [comment]
Hi,

I would recomment you use [ `torch.nn.utils.stateless`](https://pytorch.org/docs/stable/generated/torch.nn.utils.stateless.functional_call.html#torch.nn.utils.stateless.functional_call)
to be able to easily call Modules in a functional way!

**Comment by albanD** [comment]
You can do it like this:
```python
regular_out = rnn(input, (h0, c0))

state = rnn.state_dict()
functional_out = nn.utils.stateless.functional_call(rnn, state, (input, (h0, c0)))
```

**Comment by snow2white** [comment]
Great, thanks!

**Comment by albanD** [comment]
cc @jbschlosser  I think we can close this as won't fix right?

**Comment by mariaalfaroc** [comment]
I have another question. Let's say I'm using a Hypernetwork to predict the weights of such RNN. If both networks are on GPU, I get an error when trying to compute gradients: "cudnn RNN backward can only be called in training mode".
[Colab for reproducible error.](https://colab.research.google.com/drive/17TkUfv2D6C4E63wjD4l1kxjQpLcq2wur?usp=sharing)
How can I solve this?

**Comment by mariaalfaroc** [comment]
I'm using the model in evaluation mode because I want it to make predictions according to the weights predicted by the Hypernetwork, meaning that I'm only training the Hypernetwork. If I remove the `.eval(),` then the Dropout layers would be used during the stateless call (I can remove then and problem solved) but I worry that the BatchNormalization layers would not worked as when they do in evaluation mode. Do BatchNorm layers work in the same way as in evaluation mode when they are called by `torch.nn.utils.stateless`?

**Comment by albanD** [comment]
It will use whatever is the current state of the module expect the things you pass in. So if the module is in eval mode, that will be used!

**Comment by mariaalfaroc** [comment]
Okey! Thanks :)
Then, I think I will have to use `torch.backends.cudnn.enabled=False`. Let's say the decoder is the RNN module. To disable cudnn around that module it would be like this?

```
def forward(self, x):
        x = self.encoder(x)
        torch.backends.cudnn.enabled = False
        x = self.decoder(x)
        torch.backends.cudnn.enabled = True
        return x
```

**Comment by albanD** [comment]
Yep that will work.
